src/collection/cereal.rs

```diff
@@ -62,6 +68,64 @@ where
     Ok(map)
 }
 
+// Custom deserializer for `Template`. This is useful for deserializing values
+// that are not strings, but should be treated as strings such as numbers,
+// booleans, and nulls.
+impl<'de> Deserialize<'de> for Template {
+    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+    where
+        D: Deserializer<'de>,
+    {
+        struct TemplateVisitor;
+
+        macro_rules! visit_primitive {
+            ($func:ident, $type:ty) => {
+                fn $func<E>(self, v: $type) -> Result<Self::Value, E>
+                where
+                    E: Error,
+                {
+                    Template::try_from(v.to_string()).map_err(E::custom)
+                }
+            };
+        }
+
+        macro_rules! visit_null {
+            ($func:ident) => {
+                fn $func<E>(self) -> Result<Self::Value, E>
+                where
+                    E: Error,
+                {
+                    Template::try_from("null".to_string()).map_err(E::custom)
+                }
+            };
+        }
+
+        impl<'de> Visitor<'de> for TemplateVisitor {
+            type Value = Template;
+
+            fn expecting(
+                &self,
+                formatter: &mut std::fmt::Formatter,
+            ) -> std::fmt::Result {
+                formatter.write_str(
+                    "Invalid type, must be a string, number, boolean, or null",
+                )
+            }
+
+            visit_primitive!(visit_bool, bool);
+            visit_primitive!(visit_u64, u64);
+            visit_primitive!(visit_i64, i64);
+            visit_primitive!(visit_f64, f64);
+            visit_primitive!(visit_str, &str);
+
+            visit_null!(visit_none);
+            visit_null!(visit_unit);
```

Comment:
I think supporting `null` here could potentially be confusing, and treating unit as `null` is especially confusing. Leaving an empty value for a key and having it populate as `"null"` is unintuitive. Having `null` populate as `"null"`  is maybe a bit more intuitive but I still don't like it because it ties us to the YAML format.

My preference would be to have `null` not deserialize at all (so it gives an error), and unit deserializes as the empty string.

I'm not sure if this was in your original revision but if so, sorry I missed it

Reply:
My bad, I should have only done booleans and numbers. I think there should be separate issue regarding null value.